slsDetectorSoftware/src/slsDetector.cpp

```diff
@@ -2741,25 +2741,11 @@ void slsDetector::setReceiverDbitList(std::vector<int> list) {
     shm()->rxDbitListSize = list.size();
     std::copy(list.begin(), list.end(), shm()->rxDbitList);
  
-    int ret = FAIL;
     if (shm()->rxOnlineFlag == ONLINE_FLAG) {
-        int fnum = F_SET_RECEIVER_DBIT_LIST;
-        int arg = list.size();
-        auto receiver =
-            sls::ClientSocket("Receiver", shm()->rxHostname, shm()->rxTCPPort);
-        receiver.sendData(&fnum, sizeof(fnum));
-        receiver.sendData(&arg, sizeof(arg));
-        receiver.sendData(shm()->rxDbitList, arg * sizeof(int));
-        receiver.receiveData(&ret, sizeof(ret));
-        if (ret == FAIL) {
-            char mess[MAX_STR_LENGTH]{};
-            receiver.receiveData(mess, MAX_STR_LENGTH);
-            throw ReceiverError("Receiver " + std::to_string(detId) +
-                                " returned error: " + std::string(mess));
-        }
-    }
-    if (ret == FORCE_UPDATE) {
-        ret = updateCachedReceiverVariables();
+        int args[list.size() + 1];
+        args[0] = list.size();
+        std::copy(std::begin(list), std::end(list), args + 1);
+        sendToReceiver(F_SET_RECEIVER_DBIT_LIST, args, sizeof(args), nullptr, 0);
```

Comment:
Here you also don't need to set the size:
sendToReceiver(F_SET_RECEIVER_DBIT_LIST, args, nullptr);
Should be enought

Reply:
I had tried it earlier, but it doesnt work when it is an array

Reply:
Could you please try again. This should work. I have done it on many places
sendToReceiver(F_SET_RECEIVER_DBIT_LIST, args, nullptr);

But if you add a ref on args it won't work since the size will be wrong. 
sendToReceiver(F_SET_RECEIVER_DBIT_LIST, &args, nullptr);

Reply:
I tried without the ref on args.
sendToReceiver(F_SET_RECEIVER_DBIT_LIST, args, nullptr);
And it doesnt compile.